src/components/MessagesList/MessagesGroup/Message/Message.vue

```diff
@@ -704,13 +695,22 @@ export default {
 			}
 		},
 
-		highlightAnimation() {
+		// triggered by custom event 'highlight' from MessagesList
+		highlightAnimationStart() {
 			// trigger CSS highlight animation by setting a class
 			this.$refs.message.classList.add('highlight-animation')
+			this.$refs.message.addEventListener('animationend', this.highlightAnimationStop)
+			// TODO need to trigger click outside of unified search to close it
+			// and then focus message
+			this.$refs.message.click()
+			setTimeout(() => {
+				this.$refs.message.focus()
+			}, 0)
```

Comment:
After closing the unifiied search popover, focus automatically moves to this popover's toggle button.

Question here is: do we need to move native focus to move from unified search to message itself? Because just highlighting seems to be not enough as per issue requirements, and native focusing force popover to close, which is a UX change

Reply:
> Question here is: do we need to move native focus to move from unified search to message itself? Because just highlighting seems to be not enough as per issue requirements, and native focusing force popover to close, which is a UX change

In the best scenario I'd say we should have all together:
- keeping search open during the search
- native focus to the selected message
- ability to return back to the search and move between messages after the search

For example, by focus selected message after search close. Or by disabling focus trap in some scenarios. But I'm not sure it is possible with unified search easily.

If choose, I'd prefer to have open search even there is no native focus on the message.

But we can discuss it with the team.

Reply:
For infromation: UnifiedSearch has an NcHeaderMenu under the hood, which uses focus-trap: https://github.com/nextcloud/nextcloud-vue/blob/master/src/components/NcHeaderMenu/NcHeaderMenu.vue

I reverted forced closing of it for now, to not mess with the behavior too much. Messages are now properly appearing on screen and highlighted, so only the issue with keyboard accessibility of this message is not solved yet => follow-up?